### ISSUE TITLE
Built CSS and JS and HTML should be minified

### ISSUE THREAD

**Issue by akrawchyk** [istrategylabs/mo-static#5]
As a developer, I want to minify my built CSS and JS and HTML, so I can increase FE performance by reducing page load.

This relates to #4 such that one possibility of minifying these assets is during the version step, see https://github.com/jamesknelson/gulp-rev-replace#usage.

We could also do them in sequence, first cache-busting the files then minifying them.

* Use https://github.com/terinjokes/gulp-uglify to mimify JS.
* Use https://github.com/murphydanger/gulp-minify-css to minify CSS.
* Use https://github.com/jonschlinkert/gulp-htmlmin to minify HTML. We'll have to set some special options for this to get maximum butter, use:
```
{
    collapseWhitespace: true,
    removeComments: true,
    removeRedundantAttributes: true,
    removeScriptTypeAttributes: true,
    removeStyleLinkTypeAttributes: true
}
```

**Comment by josephabrahams** [comment]
Does `removeScriptTypeAttributes` keep non-`text/javascript` type attributes (i.e. `text/jsx`)?

**Comment by akrawchyk** [comment]
@josephabrahams nope, [from the docs](https://github.com/kangax/html-minifier#options-quick-reference):

> Remove type="text/javascript" from script tags. Other type attribute values are left intact.

**Comment by thomasdegry** [comment]
PR open: https://github.com/istrategylabs/mo-static/pull/8

**Comment by josephabrahams** [comment]
Also, can we remove {{ cookiecutter.public_path }} and just default to `/public`?

**Comment by thomasdegry** [comment]
@josephabrahams yes and no, right now cookiecutter asks for `public path` when creating a new project, which defaults to `/public`, but you have the option to change that... So if we want `{{ cookiecutter.public_path }}` form the gulpfile we can't prompt for public path anymore

**Comment by thomasdegry** [comment]
So reading you want to default to `/public` yes we can, should I go for it?

**Comment by thomasdegry** [comment]
Updated https://github.com/istrategylabs/mo-static/pull/8

**Comment by akrawchyk** [comment]
Fixed with #8
